S3Common.h

```diff
@@ -0,0 +1,88 @@
+#if !defined(S3Common_h)
+#define S3Common_h
+
+#include <chrono>
+#include <memory>
+#include <mutex>
+#include <ostream>
+
+// libs3.h
+struct S3BucketContext;
+
+namespace cce::tf {
+class S3LibWrapper;
+class S3Connection;
+typedef std::shared_ptr<S3Connection> S3ConnectionRef;
+
+class S3Request {
+  public:
+    enum class Type {undef, get, put};
+    enum class Status {waiting, ok, error};
+    typedef std::function<void(S3Request*)> Callback;
+    static constexpr std::chrono::milliseconds max_timeout{60000};
+
+    const Type type;
+    const S3BucketContext* bucketCtx;
```

Comment:
Because it is passed as a bare pointer into s3lib (to serve as a context struct for callbacks, very C-like) I allocate with `new` on https://github.com/hep-cce2/root_serialization/blob/6f36afd39719f35b25c855e3e08b8e8aeea38964/S3Common.cc#L43-L44 and https://github.com/hep-cce2/root_serialization/blob/6f36afd39719f35b25c855e3e08b8e8aeea38964/S3Common.cc#L53-L54, and then delete once the request is finalized on https://github.com/hep-cce2/root_serialization/blob/6f36afd39719f35b25c855e3e08b8e8aeea38964/S3Common.cc#L163-L164